x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/query_performance/query_performance.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useValues } from 'kea';
+
+import { EuiBadge, EuiBadgeProps } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { ResultSettingsLogic } from '../result_settings_logic';
+
+enum QueryPerformanceRating {
+  Optimal = 'Optimal',
+  Good = 'Good',
+  Standard = 'Standard',
+  Delayed = 'Delayed',
+}
+
+type QueryPerformanceBadgePropsCollection = {
+  [x in QueryPerformanceRating]: EuiBadgeProps;
+};
+
+type QueryPerformanceBadgeContentCollection = {
+  [x in QueryPerformanceRating]: string;
+};
+
+const QueryPerformanceBadgeProps: QueryPerformanceBadgePropsCollection = {
+  [QueryPerformanceRating.Optimal]: {
+    color: '#59deb4',
+  },
+  [QueryPerformanceRating.Good]: {
+    color: '#40bfff',
+  },
+  [QueryPerformanceRating.Standard]: {
+    color: '#fed566',
+  },
+  [QueryPerformanceRating.Delayed]: {
+    color: '#ff9173',
+  },
```

Comment:
[not a change request, just a thought for the redesign] @daveyholler - should we put in a tech debt item at some point to evaluate these badge colors for accessibility?

Reply:
@JasonStoltz separately, if we're only setting the `color` prop, why bother returning an obj instead of a color string? can we simplify this like so:

```js
const QueryPerformanceColor: = {
  [QueryPerformanceRating.Optimal]: '#59deb4',
  // ... etc
}

<EuiBadge
  color={QueryPerformanceColor[queryPerformanceRating]}
/>

Reply:
Good point Constance, this was just carry over from the migration. I can update it.

Reply:
cab6ee6dc9e5d2492b04478d57aee860d7145daa